subsys/debug/coredump/coredump_core.c

```diff
@@ -158,3 +158,29 @@ void z_coredump_memory_dump(uintptr_t start_addr, uintptr_t end_addr)
 
 	z_coredump_buffer_output((uint8_t *)start_addr, len);
 }
+
+int coredump_query(enum coredump_query_id query_id, void *arg)
+{
+	int ret;
+
+	if (backend_api->query == NULL) {
+		ret = -ENOTSUP;
+	} else {
+		ret = backend_api->query(query_id, arg);
+	}
+
+	return ret;
```

Comment:
just a thought - we could condense this, unless we see it scaling up to different conditional statements. 
`return (backend_api->query == NULL) ? (-ENOTSUP) : (backend_api->query(query_id, arg));`

Reply:
the code as it is is easier to read, you can add comment, if it changes, you it is easier to review in form of a patch, and above all, there is no gain in making this a one liner.

Reply:
ok, understood. it was just a thought.